packages/xstate-inspect/src/browser.ts

```diff
@@ -168,6 +168,10 @@ export function inspect(
     };
 
     service.subscribe((state) => {
+      // filter out synchronous notification from within `.start()` call when the `service.state` has not yet been assigned
+      if (state === undefined) {
+        return;
+      }
```

Comment:
A more semantic fix would be to add a local variable like `skipSyncNotification = true`, filter based on it, and assign it to `false` right after `service.subscribe` returns. 

It would look quite weird though so I've kept the initial condition that I've written - let me know if you'd like the other one to get introduced. This kinda showcases how calling the listener synchronously on the subscription time is often problematic where the intention is to react to changes.

Reply:
This fix is fine for now; we can reinvestigate later